airflow/models/dagwarning.py

```diff
@@ -79,7 +83,12 @@ def purge_inactive_dag_warnings(cls, session: Session = NEW_SESSION) -> None:
             query = session.query(cls).filter(cls.dag_id.in_(dag_ids))
         else:
             query = session.query(cls).filter(cls.dag_id == DagModel.dag_id, DagModel.is_active == false())
-        query.delete(synchronize_session=False)
+        try:
+            query.delete(synchronize_session=False)
+        except OperationalError:
+            # If the purge query fails, it's not critical to the dag processor
+            # We ignore it so the dag processor manager doesn't exit
+            log.error("Failed to purge inactive dag_warnings, ignoring")
```

Comment:
You can instead re-try the transaction https://github.com/apache/airflow/blob/23264fb820c179e9951ea9706f68b13a9b3fdbc0/airflow/utils/retries.py#L49

Reply:
What do you think of this? Also, do you think we need to run this query every parsing loop? What if we only run it every x seconds and make that configurable?